xbmc/cores/VideoPlayer/DVDSubtitles/DVDFactorySubtitle.cpp

```diff
@@ -51,8 +52,10 @@ CDVDSubtitleParser* CDVDFactorySubtitle::CreateParser(std::string& strFile)
       {
         return new CDVDSubtitleParserVplayer(std::move(pStream), strFile.c_str());
       }
-      else if ((!memcmp(line, "Dialogue: Marked", 16)) || (!memcmp(line, "Dialogue: ", 10)) ||
-               (!memcmp(line, "[Events]", 8)))
+      else if (!StringUtils::CompareNoCase(line, "!: This is a Sub Station Alpha v1", 33) ||
```

Comment:
Is this maybe a little to strict a search.
According to https://wiki.multimedia.cx/index.php?title=SubStation_Alpha there is also V2,3,4 for SSA, with ASS being 4+.
I can't really say what the likelihood of using the various versions are, so maybe it doesn't matter in real terms, not sure.

Reply:
in order to detect at first line all old versions (1-2-3)
i can change `!: This is a Sub Station Alpha v1` with `!: This is a Sub Station Alpha v` no problem
but the headers are not always included in the ASS/SSA file then fallback in to the other cases

the versions 4 and 4+ are both detected by `ScriptType: v4.00`
already tested